### PR TITLE
Add documentation about OLM installation

### PR DESCRIPTION
**Pull request by clamoriniere**
### What does this PR do?

Update documentation with Operator Lifecycle Manager instruction.

### Motivation

Explain how to override the default Operator pod's resources

### Additional Notes

N/A

### Describe your test plan

Install the operator with OLM, try to used the example `Subscription`. The Pod resources should be used.

### CODE REVIEW

**Review by vboulineau — APPROVED** [review]

File: docs/installation.md [hunk 1]
```diff
@@ -41,11 +43,42 @@ To use the Datadog Operator, deploy it in your Kubernetes cluster. Then create a
    helm install $DD_NAMEOP -n $DD_NAMESPACE .
    ```
 
+### With the Operator Lifecycle Manager
+
+The Datadog Operator deployment with [Operator Lifecycle Manager][7] documentation is available on the [operatorhub.io][8].
```
Comment (ruthnaebeck) [inline docs/installation.md:48]:
```suggestion
The Datadog Operator deployment with [Operator Lifecycle Manager][7] documentation is available at [operatorhub.io][8].
```

File: docs/installation.md [hunk 2]
```diff
@@ -41,11 +43,42 @@ To use the Datadog Operator, deploy it in your Kubernetes cluster. Then create a
...
+
+#### Override default Operator configuration
+
+The [Operator Lifecycle Manager][7] framework allows overriding default Operator configuration. The list of the supported installation configuration parameters is documented [here][9].
```
Comment (ruthnaebeck) [inline docs/installation.md:52]:
```suggestion
The [Operator Lifecycle Manager][7] framework allows overriding default Operator configuration. See the [Subscription Config][9] document for a list of the supported installation configuration parameters.
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Docs review

**Comment by clamoriniere** [comment]
Hi @ruthnaebeck,

Thank you for the review. I have updated the PR with your suggestions.

**Review by ruthnaebeck — APPROVED** [review]
👍 for docs

**Pull request merged**